packages/core/src/lib/TwaManifest.ts

```diff
@@ -372,15 +371,18 @@ export class TwaManifest {
     if (!(fieldsToIgnore.includes('shortcuts'))) {
       shortcuts = this.getShortcuts(webManifestUrl, webManifest);
     }
-    const icon = (fieldsToIgnore.includes('icons'))? null :
-        findSuitableIcon(webManifest.icons, 'any', MIN_ICON_SIZE);
-    const maskableIcon = (fieldsToIgnore.includes('maskableIcons'))? null :
-        findSuitableIcon(webManifest.icons, 'maskable', MIN_ICON_SIZE);
-    const monochromeIcon = (fieldsToIgnore.includes('monochromeIcons'))? null :
-        findSuitableIcon(webManifest.icons, 'monochrome', MIN_NOTIFICATION_ICON_SIZE);
+    const oldTwaManifestJson = oldTwaManifest.toJson();
+    const iconUrl = (fieldsToIgnore.includes('icons'))? oldTwaManifestJson.iconUrl:
+       resolveIconUrl(findSuitableIcon(webManifest.icons, 'any', MIN_ICON_SIZE));
+    const maskableIconUrl = (fieldsToIgnore.includes('maskableIcons'))?
+        oldTwaManifestJson.maskableIconUrl :
+        resolveIconUrl(findSuitableIcon(webManifest.icons, 'maskable', MIN_ICON_SIZE));
+    const monochromeIconUrl = (fieldsToIgnore.includes('monochromeIcons'))?
+        oldTwaManifestJson.monochromeIconUrl :
+        resolveIconUrl(findSuitableIcon(webManifest.icons, 'monochrome',
+            MIN_NOTIFICATION_ICON_SIZE));
```

Comment:
I feel these three statements duplicate a fair amount of code and I think we could extract them to another method (similar to `getNewFieldValue`) (although it would probably take a horrible amount of parameters), something like:

```
function getNewIconUrl(fieldsToIgnore, fieldName, oldUrl, icons, iconType, iconSize);
```

(feel free to order the parameters as you feel appropriate)